api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/CXCASCRNCohortQueries.java

```diff
@@ -77,6 +78,392 @@ private CohortDefinition getA() {
+            + "        AND        e.voided = 0 "
+            + "        AND        o.voided = 0 "
+            + "        AND        o2.voided = 0 "
+            + "        AND        e.encounter_type = ${28} "
+            + "        AND        ( ( o.concept_id = ${2094} "
+            + "        AND            o.value_coded IN (${2093}, ${664}, ${703}) ) "
+            + "         OR          (  o2.concept_id = ${165436} "
+            + "        AND             o2.value_coded = ${664} ) )  "
+            + "        AND        e.location_id = :location "
+            + "        AND        e.encounter_datetime  ";
+    if (beforeStartDate) {
+      query += (" < :startDate ");
+    } else {
+      query += ("  BETWEEN :startDate AND  :endDate ");
+    }
+
+    StringSubstitutor sb = new StringSubstitutor(map);
+
+    cd.setQuery(sb.replace(query));
+
+    return cd;
+  }
+
+  /**
+   * <b>SCRN_FR9</b> Numerator Disaggregation: Result
+   *
+   * <ul>
+   *   <li>Negative
+   *       <ul>
+   *         <li>All screened patients (SCRN_FR4) who have been marked as VIA Negativo on Resultado
+   *             VIA OR have Resultado do Rastreio HPV-DNA marked as Negativo
+   *       </ul>
+   * </ul>
+   *
+   * <p><b>Note:</b> If there is more than one HPV result or VIA result registered during the
+   * reporting period, the system will consider the most recent result.<br>
+   * <br>
+   *
+   * <p><b>Note:</b> If there are discrepant results registered on the same (most recent) day, the
+   * system will consider the following hierarchy:
+   *
+   * <ul>
+   *   <li>Positive
+   *   <li>Suspected Cancer
+   *   <li>Negative
+   * </ul>
+   */
+  public CohortDefinition getPatientsWithNegativeResultForScreeningTest(boolean beforeStartDate) {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+    cd.setName("Disagregation - Patients with Negative Screening Result");
+    cd.addParameter(new Parameter("startDate", "startDate", Date.class));
+    cd.addParameter(new Parameter("endDate", "endDate", Date.class));
+    cd.addParameter(new Parameter("location", "location", Location.class));
+
+    Map<String, Integer> map = new HashMap<>();
+    map.put("28", hivMetadata.getRastreioDoCancroDoColoUterinoEncounterType().getEncounterTypeId());
+    map.put("2094", hivMetadata.getResultadoViaConcept().getConceptId());
+    map.put("664", hivMetadata.getNegative().getConceptId());
+    map.put("703", hivMetadata.getPositive().getConceptId());
+    map.put("2093", hivMetadata.getSuspectedCancerConcept().getConceptId());
+    map.put("165436", hivMetadata.getHumanPapillomavirusDna().getConceptId());
+
+    String query =
+        "SELECT patient_id "
+            + "FROM   (SELECT negative.patient_id, "
+            + "               Max(negative.last_negative_result_date) AS negative_result_Date "
+            + "        FROM   (SELECT p.patient_id, "
+            + "                       Max(e.encounter_datetime) AS last_negative_result_date "
```

Comment:
Choose one of the clauses to pick the max date, I recommend to pick on the first one (outer)

Reply:
Done